apps/messaging-app-extn-playground/api/util.js

```diff
@@ -0,0 +1,141 @@
+        headers: {
+            Authorization: `Bearer ${accessToken}`,
+        },
+        json: true,
+    };
+    const userInfo = await request(requestOptions);
+
+    return userInfo;
+}
+
+// Create a messaging channel in Pipedrive
+async function createChannel(accessToken, id, name, type) {
+
+    const requestOptions = {
+        uri: "https://api.pipedrive.com/v1/channels",
+        method: "POST",
+        headers: {
+            Authorization: `Bearer ${accessToken}`,
+        },
+        body: {
+            name: name,
+            provider_channel_id: id,
+            avatar_url: 'https://robohash.org/mxtouwlpxqjqtxiltdui?set=set1&bgset=&size=48x48',
+            provider_type: type
+        },
+        json: true,
+    };
+    const response = await request(requestOptions);
+    debug("Channel created!");
+
+    return response;
+}
+
+// Forward message to WhatsApp using their Graph API
+async function sendMessageToWA(msg, recipientId) {
+    debug("Sending a whatsapp message based on data received from Pipedrive");
+    const requestOptions = {
+        uri: `https://graph.facebook.com/v13.0/${process.env.WA_PHONE_NUMBER_ID}/messages`,
+        method: "POST",
+        headers: {
+            Authorization: `Bearer ${process.env.WA_TOKEN}`,
+        },
+        json: {
+            messaging_product: "whatsapp",
+            to: recipientId.split('wa-')[1],
+            type: "text",
+            text: {
+                preview_url: true,
+                body: msg,
+            },
+        },
+    };
+    const response = await request(requestOptions);
+    debug("Message sent to WhatsApp from Pipedrive");
+    return response;
+}
+
+// Forward message to Pipedrive Messaging Inbox using Channels API
+async function sendMessageToPD(accessToken, from, msg, time) {
+    debug("Sending a Pipedrive Inbox message based on data from WhatsApp chat:", msg);
+    const requestOptions = {
+        uri: "https://api.pipedrive.com/v1/channels/messages/receive",
+        method: "POST",
+        headers: {
+            Authorization: `Bearer ${accessToken}`,
+        },
+        body: {
+            id: "msg-wa-" + Date.now(),
+            channel_id: process.env.CHANNEL_ID,
+            conversation_id: `conversation-${from}`,
+            sender_id: `sender-wa-${from}`,
+            message: msg,
+            status: "sent",
+            created_at: new Date(parseInt(time) * 1000).toISOString().replace("T", " ").substring(0, 16),
+            attachments: [],
+        },
+        json: true,
+    };
+    const status = await request(requestOptions);
+    debug("Message sent to Pipedrive from WhatsApp");
+    return status;
+}
+
+// Figure out the domain in which the app is running
+async function getAppDomain(port = 3000) {
+    let domain;
+    if (process.env.PROJECT_DOMAIN) {
+        domain = `https://${process.env.PROJECT_DOMAIN}`;
+    } else {
+        domain = `http://localhost:${port}`;
+    }
+    return domain;
+}
```

Comment:
This function probably doesn't have to be async unless you're leaving it async so in future it could hold asynchronous code?

Reply:
Good point. It will not have async code. Makes sense to keep it normal